src/index.js

```diff
@@ -129,11 +129,40 @@ for (let i=ELEMENTS.length; i--; ) {
 }
 
 
+function isStatelessComponent(c) {
+	return typeof c === 'function' && !(c.prototype instanceof Component);
+}
+
+
+function statelessComponentHook(WrappedComponent) {
+	// Ensure the displayName has been set for debugging purposes
+	WrappedComponent.displayName = WrappedComponent.displayName
+		|| WrappedComponent.name
+		|| 'StatelessComponent';
+
+	function StatelessComponent(props) {
+		// Ensure default props have been applied to the props.
+		const p = WrappedComponent.defaultProps
```

Comment:
I believe defaultProps should already be applied by this point - stateless functions are invoked [here](https://github.com/developit/preact/blob/master/src/vdom/functional-component.js#L24), and their props are mutated prior to invocation [here](https://github.com/developit/preact/blob/master/src/vdom/index.js#L36-L45), which seems to apply `defaultProps`.

Thoughts?